ceph.go

```diff
@@ -166,19 +168,24 @@ func (pqo *pgQueryOut) getCompletePeers() []int {
 			if err != nil {
 				panic(fmt.Sprintf("%s: %s in peer ID %q is not a valid index", pqo.Info.PgID, m[2], pi.Peer))
 			}
+
+			// Save the last_epoch_clean for later comparison
+			osdEpochMap[osd] = pi.Stats.LastEpochClean
+
 			if peers[index] == osd {
 				continue
 			}
 			if peers[index] != invalidOSD {
-				// This almost certainly means that a backfill
-				// completed and the stale shard hasn't been
-				// cleaned up yet, but we haven't been able to
-				// observe a PG query from this case yet.
-				fmt.Printf("WARNING: PG %s has multiple complete shards at index %d - current acting or first queried chosen as upmap target", pqo.Info.PgID, index)
-				continue
+				// Choose the shard with the newest last_epoch_clean
+				if osdEpochMap[peers[index]] > pi.Stats.LastEpochClean {
+					continue
+				}
 			}
 			peers[index] = osd
 		} else {
+			if pi.Incomplete == 1 {
+				continue
+			}
```

Comment:
Yeah, turns out it is not a useful method for determining the correct osds for ec

Reply:
It might be worth a comment here explaining a logic - that we're simply picking all of the complete peers for replicated since there's no way to know which ones should replace others, like we do for EC. Something like that?